tests/globus_test.py

```diff
@@ -0,0 +1,394 @@
+
+                    # request checksum again and make sure it returns quickly since it is cached in metadata
+                    time1 = time.time()
+                    session.write(get_telnet_checksum_command(checksum_type, filename1))
+                    checksum = parse_telnet_checksum_response_as_string(session.read_until(b'\r\n'))
+                    time2 = time.time()
+                    self.assertEqual(checksum, value, f'{checksum_type} checksums are not equal')
+                    self.assertLess(time2 - time1, 1, f'{checksum_type} checksums took too long to retrieve')
+
+            # overwrite filename1 with filename2
+            with FTP() as ftp:
+                ftp.connect(host='localhost', port=2811)
+                ftp.login(user='user1', passwd='pass')
+                with open(filename2,'rb') as f:
+                    ftp.storbinary(f'STOR /tempZone/home/user1/{filename1}', f)
+
+            # get and verify checksums.  the cached value in metadata should be discarded and checksums recalculated
+            with telnetlib.Telnet('localhost', 2811, 100) as session:
+                session.read_until(b'ready.\r\n')
+                session.write(b'user user1\r\n')
+                session.read_until(b'Password required for user1.\r\n')
+                session.write(b'pass\r\n')
+                session.read_until(b'User user1 logged in.\r\n')
+
+                for checksum_type, value in checksum_map_filename2.items():
+                    # send checksum request via telnet, read response and validate checksums
+                    session.write(get_telnet_checksum_command(checksum_type, filename1))
+                    checksum = parse_telnet_checksum_response_as_string(session.read_until(b'\r\n'))
+                    self.assertEqual(checksum, value, f'{checksum_type} checksums are not equal')
+
+        finally:
+            os.remove(filename1)
+            os.remove(filename2)
+
+            # remove file via ftp 
+            with FTP() as ftp:
+                ftp.connect(host='localhost', port=2811)
+                ftp.login(user='user1', passwd='pass')
+                ftp.delete(f'/tempZone/home/user1/{filename1}')
+
+    def test_checksum_with_progress_markers(self):
+        filename = f'{inspect.currentframe().f_code.co_name}1'
+
+        try:
+            # file must be big enough so that checksum takes at least one second
+            # otherwise this test will hang waiting for checksum marker
+            make_arbitrary_file(filename, 600*1024*1024)
+
+            # get various checksum of the files
+            md5 = hashlib.md5(open(filename,'rb').read()).hexdigest()
+
+            # Put a file to user1's home directory
+            # user1 is defined as the anonymous ftp user
+            # At this point I don't know how to put a file to user1 with globus-url-copy.
+            # I also don't know how to telnet to a non-anonymous ftp user which I need for checksums.
```

Comment:
Are these important/needed for this test? Should we make issues for these?

Reply:
I don't think it is important necessarily.  That is more in line with how Globus does authentication via certificates.  It would make the tests cleaner though.  I'll add an issue on it but as far as I can tell it isn't a big priority.

Reply:
Added an issue on this.

Reply:
Please link it

Reply:
https://github.com/irods/irods_client_globus_connector/issues/97